internal/subshell/bash/bash.go

```diff
@@ -131,6 +140,20 @@ func (v *SubShell) SetupShellRcFile(targetDir string, env map[string]string, nam
 
 // SetEnv - see subshell.SetEnv
 func (v *SubShell) SetEnv(env map[string]string) {
+	for key, value := range env {
+		var bashified string
+		var err error
+		if key == "PATH" {
+			bashified, err = osutils.BashifyPathEnv(value)
+		} else {
+			bashified, err = osutils.BashifyPath(value)
```

Comment:
This would cause us to attempt bashifying every single env var? I can't see that not causing issues. 

Perhaps we are over-complicating this. What if we just run `bash -c printenv` with our windows env passed in, then the result would be the completely bashified bash env, at least in theory.

Reply:
It *attempts* to bashify every single env var that matches the "C:" pattern for filenames on Windows. This is because there are other environment variables declared by the runtime that may need to be bashified (e.g. PYTHONPATH). If the env var cannot be bashified, an error is thrown and ignored.

Reply:
After thinking about this some more, perhaps only PATH needs to be bashified. Presumably any other utilities know they're running on Windows and handle paths properly.

Reply:
Okay, I was overcomplicating this. As mentioned in a previous comment, I'm doing a simple string translation of PATH only, because we only need it to help bash find runtime executables.

Reply:
I don't think that's true. You're right in that any env var can contain path values. And while we ourselves might only rely on PATH, the user may have scripts or other automations that rely on the values in the other env variables.

What about my `bash -c printenv` suggestion? That seems to me to be a silver bullet solution.

Reply:
Note I haven't yet reviewed the string translation you refer to as I'm writing this; but that sounds like a red flag to me. There's no way I can think of that we can know what a windows path looks like in a bash environment. The mounting point can differ depending on their bash setup.

Reply:
`bash -c printenv` does not do any translations/bashification. I still see "C:\Users\username" and similar paths. I'm not sure what you were expecting?

Reply:
I'm not seeing this? 
![image](https://user-images.githubusercontent.com/278649/199832218-f1ae8efd-3d5c-4e2a-ab6f-0d50c50bb358.png)

Reply:
I'm using Git Bash, which gives me this: 
![VirtualBox_Windows 10_03_11_2022_17_41_53](https://user-images.githubusercontent.com/12902397/199840338-9fe3c178-d5ba-4562-b71d-c849a4aca3ed.png)

It looks like your Bash environment is doing something extra.

Reply:
As we discovered last week, MinGW/MSYS2/WSL is auto-translating %PATH% to a bashified equivalent, but the State Tool is writing an unbashified form to a shell RC file for later sourcing, which does not take advantage of this auto-translation. We're now running `bash -c "echo \"$PATH\"` in order to write the auto-bashified %PATH% to the shell rc file.